In order to avoid _Authentication is not configured for provider_ error, resources created with the azurerm provider must be defined in a separate module than resources created using the _databricks_ module as described [here](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs#data-resources-and-authentication-is-not-configured-errors) and on [GitHub](https://github.com/databrickslabs/terraform-provider-databricks/issues/267).